packages/dds/tree/src/test/shared-tree/fuzz/fuzzEditGenerators.ts

```diff
@@ -102,56 +125,195 @@ const defaultEditGeneratorOpWeights = {
 	start: 3,
 	commit: 1,
 	abort: 1,
+	synchronize: 1,
 };
 
-export const makeEditGenerator = (
+export const makeNodeEditGenerator = (
 	opWeights: EditGeneratorOpWeights = defaultEditGeneratorOpWeights,
-): AsyncGenerator<Operation, FuzzTestState> => {
+): AsyncGenerator<NodeEdit, FuzzTestState> => {
 	type EditState = FuzzTestState & TreeContext;
-	async function insertGenerator(state: EditState): Promise<FuzzInsert> {
+
+	async function setPayloadGenerator(state: EditState): Promise<FuzzSetPayload> {
 		const trees = state.testTreeProvider.trees;
 		const tree = trees[state.treeIndex];
-
 		// generate edit for that specific tree
-		const {
-			parent: path,
-			parentField: nodeField,
-			parentIndex: nodeIndex,
-		} = getRandomPlace(tree, state.random);
-		const insert: FuzzInsert = {
-			fuzzType: "insert",
-			parent: path,
-			field: nodeField,
-			index: nodeIndex,
-			value: state.random.integer(Number.MIN_SAFE_INTEGER, Number.MIN_SAFE_INTEGER),
+		const path = getExistingRandomNodePosition(tree, state.random);
+		return {
+			nodeEditType: "setPayload",
+			path,
+			value: state.random.integer(Number.MIN_SAFE_INTEGER, Number.MAX_SAFE_INTEGER),
+			treeIndex: state.treeIndex,
 		};
-		return insert;
 	}
 
-	async function deleteGenerator(state: EditState): Promise<FuzzDelete> {
+	const baseNodeEditGenerator = createWeightedAsyncGenerator<FuzzNodeEditChange, EditState>([
+		[
+			setPayloadGenerator,
+			opWeights.setPayload ?? 0,
+			({ testTreeProvider, treeIndex }) =>
```

Comment:
this looks dangerous for a couple reasons:

1. createWeightedAsyncGenerator doesn't handle 0 weight well at all. it looks like it probably throws a confusing error or infinite loops. (for that matter, it doesn't handle non-integer weights well either). all of that should be improved in the base package, but for now I'd do something safer keeping those limitations in mind
2. `createWeightedAsyncGenerator` doesn't have a good failure mode when none of the acceptance criteria pass.

Rather than put the acceptance criteria here, you should probably move it to a higher level of operation selection--e.g. when deciding if a field or node edit is appropriate

Reply:
if you want to improve any of the things mentioned in (1) feel free to do so.

Reply:
moved acceptance criteria one layer up